### PR TITLE
ENH: Use sgl_scoring_path instead of sklearn's _path_residuals

### LINKED ISSUE #29: Implement custom _path_residuals function
`SGLCV` currently relies on sklearn's `_path_residuals` but that only returns MSE. We'd like it to return both custom scorers and fitted coefs, as it does in the logistic case.

Tagging @mnarayan since this came up in conversation.

### PR DESCRIPTION
**Pull request by richford**
Resolves #29

### CODE REVIEW

File: groupyr/sgl.py [hunk 1]
```diff
@@ -510,13 +515,181 @@ def sgl_path(
...
+    coefs, alphas, n_iter = sgl_path(
+        X_train,
+        y_train,
+        l1_ratio=l1_ratio,
+        groups=groups,
+        scale_l2_by=scale_l2_by,
+        eps=eps,
+        n_alphas=n_alphas,
+        alphas=alphas,
+        Xy=Xy,
```
Comment (arokem) [inline groupyr/sgl.py:649]: If `Xy` is really `np.dot(X, y)` then it includes information from `X_train` `y_train`, but also from `X_test` `y_test`.

**Review by arokem — COMMENTED** [review]
Overall, I think this looks good. But a few small comments and questions.

File: groupyr/logistic.py [hunk 2]
```diff
@@ -1054,6 +1057,7 @@ def fit(self, X, y):
             self.coef_ = model.coef_
             self.intercept_ = model.intercept_
             self.n_iter_ = model.n_iter_
+            self.bayes_optimizer_ = None
```
[the remark this answers is not in the document]
Comment (richford) [reply groupyr/logistic.py:1060]: Good point! Resolved in [f15103c](https://github.com/richford/groupyr/pull/33/commits/f15103c638c9a7f1837f91aaf162ef8b327a13b7)

File: groupyr/sgl.py (hunk 1 above)
Comment (richford) [reply groupyr/sgl.py:655]: That's a good point. Most of the time in groupyr, `Xy` is `None`, so we don't precompute and avoid this problem. I just confirmed that the only times that `Xy` is potentially non-`None`, it is only for the determination of the alpha grid (i.e. the max and min values of alpha in the regularization path) so there's no leakage there either. I kept `Xy` in there mostly to parallel the signatures of the scoring path functions in sklearn. I added more info to the docstrings in [26f6643](https://github.com/richford/groupyr/pull/33/commits/26f6643d457b0044dc8b092b7c28b2a5e2852ffa), including a warning about train/test leakage.

Comment (arokem) [reply groupyr/sgl.py:655]: That makes sense and I think that the docstring comment is good to have, for that weird use-case we are not thinking of (and hoping we have sense to read the docstring when we do that...).

**Comment by richford** [comment]
I added some `logger.warning` statements to the scoring functions in the case that the user provides `Xy`.

**Pull request merged**